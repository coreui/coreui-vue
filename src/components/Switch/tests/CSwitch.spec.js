import { mount } from "@vue/test-utils";
import Component from "../CSwitch";

const ComponentName = 'CSwitch'
const defaultWrapper = mount(Component)
const customWrapper = mount(Component, {
  propsData: {
    variant: 'info',
    outline: 'alt',
    size: 'lg',
    shape:'3d',
    name:'myName',
    id:'myId',
    checked: true,
    value: 'checked',
    disabled: true,
    required: true,
    trueValue: 'yes',
    falseValue: 'no',
    type: 'radio',
    dataOn: 'dataOn',
    dataOff: 'dataOff'
  }
})

describe(`${ComponentName} .vue`, () => {
  it('has a name', () => {
    expect(Component.name).toMatch(ComponentName)
  })
  it('renders correctly', () => {
    expect(defaultWrapper.element).toMatchSnapshot()
  })
  it('renders correctly', () => {
    expect(customWrapper.element).toMatchSnapshot()
  })
  // Inspect the raw component options
  // it("correctly sets default props and data when created", () => {
  //   expect(defaultWrapper.props().variant).toBe('secondary')
  //   expect(defaultWrapper.props().size).toEqual(null)
  //   expect(defaultWrapper.props().checked).toBe(false)
  //   expect(defaultWrapper.props().defaultChecked).toBe(false)
  //   expect(defaultWrapper.props().value).toBe(true)
  //   expect(defaultWrapper.props().disabled).toBe(false)
  //   expect(defaultWrapper.props().required).toBe(false)
  //   expect(defaultWrapper.props().dataOn).toBe('On')
  //   expect(defaultWrapper.props().dataOff).toBe('Off')
  // });
  // it('has computed properties and methods', () => {
  //   expect(typeof Component.computed.classList).toBe('function')
  //   expect(typeof Component.computed.isChecked).toBe('function')
  // })
  // it('computed properties return correct data', () => {
  //   expect.arrayContaining(defaultWrapper.vm.classList)
  //   expect(defaultWrapper.vm.isChecked).toBe(false)
  // })
  // it('has a handleChange method', () => {
  //   expect(typeof Component.methods.handleChange).toBe('function')
  // })
  // it('has a toggle method', () => {
  //   expect(typeof Component.methods.toggle).toBe('function')
  // })
  // it('matches default props snapshot and render correctly', () => {
  //   expect(defaultWrapper.element).toMatchSnapshot()
  //   expect(defaultWrapper.is('label')).toBe(true)
  //   expect(defaultWrapper.classes()).toContain('switch')
  //   expect(defaultWrapper.find('input').classes()).toContain('switch-input')
  //   expect(defaultWrapper.find('span').classes()).toContain('switch-slider')
  // })
  //
  // it('correctly sets custom props and data when created', () => {
  //   expect(customWrapper.props().variant).toBe('info')
  //   expect(customWrapper.props().size).toEqual('lg')
  //   expect(customWrapper.props().checked).toBe(true)
  //   expect(customWrapper.props().defaultChecked).toBe(true)
  //   expect(customWrapper.props().value).toBe(false)
  //   expect(customWrapper.props().disabled).toBe(true)
  //   expect(customWrapper.props().required).toBe(true)
  //   expect(customWrapper.props().dataOn).toBe('dataOn')
  //   expect(customWrapper.props().dataOff).toBe('dataOff')
  // })
  // it('matches custom props snapshot', () => {
  //   expect(customWrapper.element).toMatchSnapshot()
  // })
})
