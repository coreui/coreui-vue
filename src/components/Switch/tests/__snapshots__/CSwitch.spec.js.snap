// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CSwitch .vue renders correctly 1`] = `
<label
  class="switch switch-secondary form-check-label"
>
  <input
    class="switch-input form-check-input"
    type="checkbox"
    value=""
  />
   
  <span
    class="switch-slider"
  />
</label>
`;

exports[`CSwitch .vue renders correctly 2`] = `
<label
  class="switch switch-label switch-lg switch-3d switch-outline-info-alt form-check-label"
>
  <input
    class="switch-input form-check-input"
    disabled="disabled"
    id="myId"
    name="myName"
    required="required"
    type="radio"
    value="checked"
  />
   
  <span
    class="switch-slider"
    data-checked="dataOn"
    data-unchecked="dataOff"
  />
</label>
`;
